tools/mason/MasonInit.chpl

```diff
@@ -72,16 +79,65 @@ proc masonInit(args) throws {
     }
 
     if dirName == '' {
-      const cwd = getEnv("PWD");
-      var name = basename(cwd);
-      const path = '.';
-      if packageName.size > 0 then name = packageName;
-      var resName = validatePackageNameChecks(path, name);
-      name = resName;
-      validateMasonFile(path, name, show);
-      var isInitialized = validateInit(path, name, true, show);
-      if isInitialized > 0 then
-      writeln("Initialized new library project: " + basename(cwd));
+      if defaultBehavior {
+        const cwd = getEnv("PWD");
+        var name = basename(cwd);
+        const path = '.';
+        if packageName.size > 0 then name = packageName;
+        var resName = validatePackageNameChecks(path, name);
+        name = resName;
+        validateMasonFile(path, name, show);
+        var isInitialized = validateInit(path, name, true, show);
+        if isInitialized > 0 then
+        writeln("Initialized new library project: " + basename(cwd));
+      } else {
+        // check if Mason.toml file and src/moduleFile is present
+        var isMasonTomlPresent = false;
+        var isSrcPresent = false;
+        var moduleName: string;
+        if isFile('./Mason.toml') then isMasonTomlPresent = true;
+        if isDir('./src') then isSrcPresent = true;
+        // parse values from TOML File && module file
+        var defaultPackageName: string;
+        var defaultVersion: string;
+        var defaultChplVersion: string;
+        if isMasonTomlPresent {
+          const toParse = open("./Mason.toml", iomode.r);
+          const tomlFile = owned.create(parseToml(toParse));
+          if tomlFile.pathExists("brick.name") then
+            defaultPackageName = tomlFile["brick"]!["name"]!.s;
+          if tomlFile.pathExists("brick.version") then
+            defaultVersion = tomlFile["brick"]!["version"]!.s;
+          if tomlFile.pathExists("brick.chplVersion") then
+            defaultChplVersion = tomlFile["brick"]!["chplVersion"]!.s;
+        }
+        if isSrcPresent {
+          var file: string = listdir("./src");
+          moduleName = file;
+        }
+       // begin interactive session and get values input by user
+        var newPackageName, newVersion, newChplVersion: string;
+        var result = beginInteractiveSession(defaultPackageName, defaultVersion, defaultChplVersion);
+        newPackageName = result[0];
+        newVersion = result[1];
+        newChplVersion = result[2];
+        // overwrite Toml File
+        validateMasonFile('.', newPackageName, show);
+        isMasonTomlPresent = true;
+        if isMasonTomlPresent {
+          if newPackageName != defaultPackageName then
+            overwriteTomlValue("name", newPackageName);
+          if newVersion != defaultVersion then
+            overwriteTomlValue("version", newVersion);
+          if newChplVersion != defaultChplVersion then
+            overwriteTomlValue("chplVersion", newChplVersion);
+        }
+        if newPackageName + '.chpl' != moduleName {
```

Comment:
If the package name and module name do not match, shouldn't we move the old module to the package name rather than remove it?

Reply:
Good point. I'm still concerned about this removing their code. Imagine if you developed some code outside of a git repo and ran `mason init` to have it get deleted.

Come to think of it, there's nothing (today) that ensures the main mason package module matches the project name / file name. I think we could:

A)
  - Still move the file to the new name (and probably should emit a message that we are doing so).
  - In the future, add a warning or error if the module name did not match the package name

B) emit an error if the given package name does not match the module name.

Reply:
For archival purposes, we chose (A) for this PR.

Reply:
> In the future, add a warning or error if the module name did not match the package name

@ankingcodes - can you file an issue for this?